### ISSUE TITLE
[Feature Request(ui)] ui/Dialog accesibility

### ISSUE THREAD

**Issue by Shinyaigeek** [nanojsx/nano#31]
Hi!
I noticed some `nano-jsx/lib/ui/Dialog`'s accessibility fault when I read nano's code, such below

- dialog was not removed when a user push `Escape` key 
- a user traversing a web site only with a keyboard cannot close dialog because `action button in Dialog` can not ne focused on traversing with keyboard.
- dialog's background should be attached with `aria-hidden` prop because this is helpful for a screen-reader user
- dialog's background should be fixed with the current scrolling position
- the first interactive element in `Dialog` should be focused when `Dialog` appear
- the interactive element which a user focused on the last time before `Dialog` appear should be focues after `Dialog` disappear

These features are implemented in `react-spectrum` or `mui` and are helpful for anyone.

I want to implement this feature into `nano-jsx/lib-ui/Dialog`, but I want to hear your opinion on this before I implement it.

**Comment by yandeu** [comment]
Feel free to add these features :)

**Comment by yandeu** [comment]
Looks great so far. Thanks for your help :)

Btw, check out the cool pattern I just added in [ui.html](https://github.com/nanojsx/nano/commit/35799d61089a81172b90b34d99e9f10b39781817). It is very powerful. I'm sure you'll like it.

```tsx
// store reference to Pages Component
let pagesRef

class Pages extends Component {
  render(page = 'home') {
    if (page === 'home') return <h1>Homepage</h1>
    else if (page === 'about') return <h1>About</h1>
    else if (page === 'more') return <h1>More</h1>
  }
}

const App = () => {
  return (
    <div id="container">

      // save reference to Pages Component
      <Pages ref={p => (pagesRef = p)} />

      // manually update the Pages Component via its reference, and pass the Navigation id to update/render
      <Navigation onClick={e => pagesRef.update(e.id)}>
        <NavigationAction label="Home" icon="/.../home.svg" active={true} />
        <NavigationAction label="About" icon="/.../about.svg" active={false} />
        <NavigationAction label="Favorites" icon="/.../heart.svg" active={false} />
      </Navigation>
    </div>
  )
}
```

**Comment by Shinyaigeek** [comment]
Oh! This Navigation is pretty cool because it can really work with UI library(`nanojsx`) to provide lightness(bundle size) and simplicity!
